### PR TITLE
[Enhancement] Introduced a timeout parameter for lake::PublishVersionRequest

### PR DESCRIPTION
**Pull request by sduzh**
This PR contains the following optimizations:
- Introduced a timeout parameter for the PublishVersionRequest to handle potential long execution time.
- Implemented cache lookup for 'publish_version' function to enhance performance.
- Avoid submitting a large number of tasks to the thread pool at once, causing the thread pool to become saturated and the task to fail.

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/service/service_be/lake_service.cpp [hunk 1]
```diff
@@ -176,7 +212,13 @@ void LakeServiceImpl::publish_version(::google::protobuf::RpcController* control
             g_publish_tablet_version_queuing_latency << (run_ts - start_ts);
 
             TRACE_COUNTER_INCREMENT("tablet_id", tablet_id);
-            auto res = lake::publish_version(_tablet_mgr, tablet_id, base_version, new_version, txns, commit_time);
+
+            StatusOr<lake::TabletMetadataPtr> res;
+            if (std::chrono::system_clock::now() < timeout_deadline) {
+                res = lake::publish_version(_tablet_mgr, tablet_id, base_version, new_version, txns, commit_time);
+            } else {
+                res = Status::TimedOut("timeout exceeded");
```
Comment (kevincai) [inline be/src/service/service_be/lake_service.cpp:220]: possibly include the timeout_deadline and timeout_ms in the status error message.

File: be/src/storage/lake/transactions.cpp [hunk 2]
```diff
@@ -36,6 +36,13 @@ StatusOr<TabletMetadataPtr> publish_version(TabletManager* tablet_mgr, int64_t t
     VLOG(1) << "publish version tablet_id: " << tablet_id << ", txns: " << JoinInts(txn_ids, ",")
             << ", base_version: " << base_version << ", new_version: " << new_version;
 
+    auto new_metadata_path = tablet_mgr->tablet_metadata_location(tablet_id, new_version);
+    auto cached_new_metadata = tablet_mgr->metacache()->lookup_tablet_metadata(new_metadata_path);
+    if (cached_new_metadata != nullptr) {
+        LOG(INFO) << "Found cached tablet metadata for version " << new_version << " of tablet " << tablet_id;
```
Comment (kevincai) [inline be/src/storage/lake/transactions.cpp:42]: this logline may be too chatty?

also this line doesn't tell what it means when the new version of meta is found from cache when doing publish

File: be/src/service/service_be/lake_service.cpp (hunk 1 above)
Comment (sduzh) [reply be/src/service/service_be/lake_service.cpp]: Updated

File: be/src/storage/lake/transactions.cpp (hunk 2 above)
Comment (sduzh) [reply be/src/storage/lake/transactions.cpp]: Updated

**Review by kevincai — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Review by wangruin — APPROVED** [review]

**Pull request merged**